subcommand/acl-init/command.go

```diff
@@ -0,0 +1,148 @@
+	flagSecretName string
+	flagInitType   string
+	flagNamespace  string
+	flagACLDir     string
+
+	k8sClient *kubernetes.Clientset
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flags.StringVar(&c.flagSecretName, "secret-name", "",
+		"Name of secret to watch for an ACL token")
+	c.flags.StringVar(&c.flagInitType, "init-type", "",
+		"ACL init target, valid values are `client` and `sync`")
+	c.flags.StringVar(&c.flagNamespace, "k8s-namespace", "",
+		"Name of Kubernetes namespace where the servers are deployed")
+	c.flags.StringVar(&c.flagACLDir, "acl-dir", "/consul/aclconfig",
+		"Directory name of shared volume where acl config will be output")
+
+	c.k8s = &k8sflags.K8SFlags{}
+	flags.Merge(c.flags, c.k8s.Flags())
+	c.help = flags.Usage(help, c.flags)
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+	if len(c.flags.Args()) > 0 {
+		c.UI.Error(fmt.Sprintf("Should have no non-flag arguments."))
+		return 1
+	}
+
+	config, err := subcommand.K8SConfig(c.k8s.KubeConfig())
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error retrieving Kubernetes auth: %s", err))
+		return 1
+	}
+
+	// Create the Kubernetes clientset
+	c.k8sClient, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error initializing Kubernetes client: %s", err))
+		return 1
+	}
+
+	// Check if the client secret exists yet
+	// If not, wait until it does
+	var secret string
+	for {
+		secret, err = c.getSecret(c.flagSecretName)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error getting Kubernetes secret: %s", err))
+		}
+		if err == nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	if c.flagInitType == "client" {
+		// Construct extra client config hcl with acl details
+		// This will be mounted as a volume for the client to use
+		var buf bytes.Buffer
+		tpl := template.Must(template.New("root").Parse(strings.TrimSpace(clientACLConfigTpl)))
+		err = tpl.Execute(&buf, secret)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error creating template: %s", err))
+			return 1
+		}
+
+		// Write the data out as a file
+		err = ioutil.WriteFile(filepath.Join(c.flagACLDir, "acl-config.json"), buf.Bytes(), 0644)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error writing config file: %s", err))
+			return 1
+		}
+	}
+
+	return 0
+}
+
+func (c *Command) getSecret(secretName string) (string, error) {
+	secret, err := c.k8sClient.CoreV1().Secrets(c.flagNamespace).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	// Extract token
+	return string(secret.Data["token"]), nil
+}
+
+func (c *Command) Synopsis() string { return synopsis }
+func (c *Command) Help() string {
+	c.once.Do(c.init)
+	return c.help
+}
+
+const synopsis = "Initialize ACLs on Consul servers."
+const help = `
+Usage: consul-k8s server-acl-init [options]
+
+  Bootstraps servers with ACLs
```

Comment:
This description seems like it applies to the other subcommand instead of this one.

Reply:
Updated.